core/worker.go

```diff
@@ -1364,16 +1385,20 @@ func (w *worker) processQiTx(tx *types.Transaction, env *environment) error {
 
 	txGas := types.CalculateQiTxGas(tx)
 
-	gasUsed := env.wo.Header().GasUsed()
+	gasUsed := env.wo.GasUsed()
 	gasUsed += txGas
 	addresses := make(map[common.AddressBytes]struct{})
 	totalQitIn := big.NewInt(0)
 	utxosDelete := make([]types.OutPoint, 0)
+	inputDenominations := make(map[uint8]uint64)
 	for _, txIn := range tx.TxIn() {
 		utxo := env.state.GetUTXO(txIn.PreviousOutPoint.TxHash, txIn.PreviousOutPoint.Index)
 		if utxo == nil {
 			return fmt.Errorf("tx %032x spends non-existent UTXO %032x:%d", tx.Hash(), txIn.PreviousOutPoint.TxHash, txIn.PreviousOutPoint.Index)
 		}
+		if utxo.Lock != nil && utxo.Lock.Cmp(env.wo.Number(w.hc.NodeCtx())) > 0 {
```

Comment:
this calculation should be based on the parent number right? not on the pending header

Reply:
No, the calculation is based on the current block that it is in, it's the same in the state processor. We could change it to check the parent number if you want, but I don't think there's a difference